### PR TITLE
chore: add the current timeout setting in the TimeoutErrorMetricsEvent

### LINKED ISSUE #81: chore: add the current timeout setting in the TimeoutErrorMetricsEvent
Because the developer can manually configure the timeout in the initialize interface, the server doesn't know if the request got the timeout error using the default 30 seconds or user setting. To detect it on the server, we need to pass the current setting used in the failed request.

### Things to do

- Pass the current timeout setting in **seconds** via labels.

E.g.
```kt
labels = mapOf("tag" to "android", "timeout" to "5"),
```

https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/model/MetricsEventData.kt#L32

### PR DESCRIPTION
**Pull request by duyhungtnn**
Should close #81 
# Changes 
- [x] Add timeout value to TimeoutException
- [x] Update TimeoutErrorMetricEvent labels for containing the timeout value

### CODE REVIEW

**Comment by duyhungtnn** [comment]
@yshrsmz please help me to take a look

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt [hunk 1]
```diff
@@ -157,80 +157,125 @@ internal fun newErrorMetricsEvent(
...
+      is BKTException.ServiceUnavailableException -> Pair(
+        MetricsEventType.SERVICE_UNAVAILABLE_ERROR,
+        MetricsEventData.ServiceUnavailableErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.TimeoutException -> {
+        this.let {
```
Comment (yshrsmz) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt:244]: I would change method signature to

```kotlin
internal fun newEventDataMetricEvent(
  error: BKTException,
  timestamp: Long,
  featureTag: String,
  appVersion: String,
  apiId: ApiId
): EventData.MetricsEvent {
```

This way smart-cast just works out-of-the-box.

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt [hunk 2]
```diff
@@ -157,80 +157,125 @@ internal fun newErrorMetricsEvent(
...
+          Pair(
+            MetricsEventType.TIMEOUT_ERROR,
+            MetricsEventData.TimeoutErrorMetricsEvent(
+              apiId = apiId,
+              labels = labels.toMutableMap().apply {
+                // https://github.com/bucketeer-io/android-client-sdk/issues/81
+                set("timeout", (it.timeoutMillis / 1000f).toString())
+              },
```
Comment (yshrsmz) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt:252]: you should convert back to `Map`, or just declare `labels` with `mutableMapOf`

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt [hunk 3]
```diff
@@ -157,80 +157,125 @@ internal fun newErrorMetricsEvent(
...
+internal fun BKTException.toEventDataMetricEvent(
+  timestamp: Long,
   featureTag: String,
+  appVersion: String,
   apiId: ApiId,
-  type: MetricsEventType,
-): MetricsEventData {
+): EventData.MetricsEvent {
   val labels = mapOf("tag" to featureTag)
+  val typeAndEvent: Pair<MetricsEventType, MetricsEventData> =
```
Comment (yshrsmz) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt:174]:
```suggestion
  val (type, event) = when (this) {
```

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt (hunk 1 above)
Comment (duyhungtnn) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt:244]: Let me try your suggestion

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt (hunk 2 above)
Comment (duyhungtnn) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt:252]: I will convert back to Map to keep it immutable.

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt (hunk 1 above)
Comment (duyhungtnn) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt]: updated in https://github.com/bucketeer-io/android-client-sdk/pull/91/commits/b30b75f1cb80136aed2d4ec70258a19f3d47d9d0

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt (hunk 2 above)
Comment (duyhungtnn) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt]: updated in https://github.com/bucketeer-io/android-client-sdk/pull/91/commits/b30b75f1cb80136aed2d4ec70258a19f3d47d9d0

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt (hunk 3 above)
Comment (duyhungtnn) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt]: updated in https://github.com/bucketeer-io/android-client-sdk/pull/91/commits/b30b75f1cb80136aed2d4ec70258a19f3d47d9d0

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt [hunk 4]
```diff
@@ -157,80 +157,124 @@ internal fun newErrorMetricsEvent(
...
+
+      is BKTException.ServiceUnavailableException -> Pair(
+        MetricsEventType.SERVICE_UNAVAILABLE_ERROR,
+        MetricsEventData.ServiceUnavailableErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.TimeoutException -> {
```
Comment (yshrsmz) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt:244]: can you remove these curly brackets?

```suggestion
      is BKTException.TimeoutException -> Pair(...
```

**Review by yshrsmz — COMMENTED** [review]
one small nit-picking, but otherwise, it looks good 👍

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt (hunk 4 above)
Comment (duyhungtnn) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt]: sure, I removed it in https://github.com/bucketeer-io/android-client-sdk/pull/91/commits/beedfddb51339fc237cbd113dd247aad17e54a94

**Review by yshrsmz — APPROVED** [review]
👍

**Pull request merged**